### PR TITLE
Add assessments to project dashboard

### PR DESCRIPTION
**Pull request by martha**
## Description

[//]: # 'remove if not applicable'
Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4234
PT ticket: https://www.pivotaltracker.com/story/show/187372090

This PR adds a new Assessments tab on the Project dashboard where we can view all assessments for the given project.

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/api/operations/project.queries.graphql [hunk 1]
```diff
@@ -123,6 +123,34 @@ query GetProjectServices(
...
+  $sortOrder: AssessmentSortOption = ASSESSMENT_DATE
+  $filters: AssessmentFilterOptions = null
+) {
+  project(id: $id) {
+    id
+    assessments(
+      limit: $limit
+      offset: $offset
+      sortOrder: $sortOrder
+      filters: $filters
```
Comment (martha) [inline src/api/operations/project.queries.graphql:139]: Wanted to note that filtering by 'Custom Assessment' type didn't work how I expected. It returned no results despite the presence of some custom assessments. @gigxz is this a bug or due to my misunderstanding?

This issue seemed to impact the existing assessments tables (client, enrollment) as well as the newly built one, so, if it's a bug I'll create a new PT bugfix ticket.

File: src/modules/assessments/assessmentColumns.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+} = {
+  date: {
+    header: 'Assessment Date',
+    render: (a) => <AssessmentDateWithStatusIndicator assessment={a} />,
+  },
+  type: {
+    header: 'Assessment Type',
+    render: (a) => formRoleDisplay(a),
+  },
+  linkedType: {
```
Comment (martha) [inline src/modules/assessments/assessmentColumns.tsx]: I did some cleanup here to follow the pattern from e.g. CLIENT_COLUMNS which I remember discussing on some recent PR. I wanted to check what you think about this `linkedType` column. It's the easiest but probably least sophisticated way to deal with the problem of different assessment tables having slightly different requirements (the 'type' column values should have link treatment on some tables, but not on others). Maybe there is a better/prettier way to do this?

File: src/modules/assessments/assessmentColumns.tsx
Comment (martha) [inline src/modules/assessments/assessmentColumns.tsx:1]: I would include this in `util.ts` in this directory, converting it to a `.tsx` file, if you'd consider that a good idea

File: src/api/operations/project.queries.graphql (hunk 1 above)
Comment (gigxz) [reply src/api/operations/project.queries.graphql:139]: It's a bug - pt story https://www.pivotaltracker.com/story/show/187400755
Thanks!

File: src/modules/assessments/assessmentColumns.tsx (hunk 2 above)
Comment (gigxz) [reply src/modules/assessments/assessmentColumns.tsx]: This seems fine to me!

File: src/modules/assessments/util.ts [hunk 3]
```diff
@@ -42,3 +44,15 @@ export const generateAssessmentPath = (
     assessmentId: assessment.id,
   });
 };
+
+export const assessmentRowLinkTo = (
+  record: ClientAssessmentType | ProjectAssessmentType,
+  clientId?: string
```
Comment (gigxz) [inline src/modules/assessments/util.ts]: This should be required since it's needed for the path

File: src/modules/assessments/assessmentColumns.tsx
Comment (gigxz) [reply src/modules/assessments/assessmentColumns.tsx:1]: Yep that sounds like a good idea

**Review by gigxz — APPROVED** [review]
🌟

**Pull request merged**